includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1021,13 +1021,13 @@ public function is_original_request( $headers ) {
 	}
 
 	/**
-	 * Create a new PaymentIntent.
+	 * Generates the request when creating a new payment intent.
 	 *
 	 * @param WC_Order $order           The order that is being paid for.
 	 * @param object   $prepared_source The source that is used for the payment.
-	 * @return object                   An intent or an error.
+	 * @return array                    The arguments for the request.
 	 */
-	public function create_intent( $order, $prepared_source ) {
+	public function generate_create_intent_request( $order, $prepared_source ) {
```

Comment:
This function has this line now:

```
'statement_descriptor' => $full_request['statement_descriptor']
```

However, the statement descriptor is not always present. So, we need something like this instead:

```
if ( isset( $full_request['statement_descriptor'] ) ) {
	$request['statement_descriptor'] = $full_request['statement_descriptor'];
}
```

Reply:
Fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/962/commits/4837f5b8ace86b01154e4353b8bac5cfb5eadce5

Reply:
> However, the statement descriptor is not always present.

I don't get it. This PR doesn't change how the `$request` is created, it just re-arranges the code a bit. So, why is this now a problem, but not before? This is a good change to make, but, does that mean that payments threw PHP notices before if there's no `statement_descriptor` configured?

Reply:
It was not a problem before, because in "real-life" usage, there was always a pair of `capture` and `statement_descriptor` present, but this is not the case with tests. And it makes more sense to be careful about the presence of those parameters, rather than adding exceptions to tests.